Add some r/w functionality to GCP Bigtable Hook

EmilyMazo opened the pull request:
Make sure you have checked _all_ steps below.

### Jira

- [ X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6155

### Description

- [ X] Here are some details about my PR, including screenshots of any UI changes:
The GCP Bigtable Hook, a wrapper about the GCP Client functionality for Bigtable, has only a few methods currently, all addressing administrative tasks. I would like to add the ability to read and write Bigtable data in rows using the Hook.

### Tests

- [X ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
tests/gcp/hooks/test_bigtable/test_create_rowkey_regex_filter
tests/gcp/hooks/test_bigtable/test_delete_row
tests/gcp/hooks/test_bigtable/test_check_and_mutate_row

### Commits

- [ X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

EmilyMazo committed:
[AIRFLOW-6155] Add some r/w functionality to GCP Bigtable Hook

The bigtable hook handles administrative actions
pertaining to GCP objects like instances and clusters,
but does not have the ability to read and write Bigtable
data. This adds limited read/write functionality in the
standard hook interface format, to atomically write
and delete rows and search for rows by row key.

turbaszek commented on airflow/gcp/hooks/bigtable.py:
```diff
@@ -276,3 +278,58 @@ def get_cluster_states_for_table(instance: Instance, table_id: str) -> Dict[str,
         table = Table(table_id, instance)
         return table.get_cluster_states()
+
+    def create_rowkey_regex_filter(self, regex):
```
Where is the helper used?

EmilyMazo replied:
This helper (and the new one like it) is only used outside the Hook, for creating RowFilter objects that can be passed in to the `check_and_mutate_row` method.

turbaszek commented on airflow/gcp/hooks/bigtable.py:
```diff
@@ -276,3 +278,58 @@ def get_cluster_states_for_table(instance: Instance, table_id: str) -> Dict[str,
+        return RowKeyRegexFilter(regex)
+
+
+    def delete_row(self, instance, table_id, row_key):
```
Please add type annotations.
```suggestion
    def delete_row(self, instance: Instance, table_id: str, row_key: str) -> None:
```

Commit:
[AIRFLOW-6155] Adding type annotations and another helper method

Adding type annotations to new methods as
well as the app_profile_id parameter to all
helper methods that use the BigtableTableClient
to send requests to a Table. Also added a helper
method for creating ValueRegexFilter objects
in the style of the other contributed Hooks.

Commit:
[AIRFLOW-6155] Type annotation for RowFilter

Adding a missing type annotation for a RowFilter parameter

Commit:
[AIRFLOW-6155] Type annotations and newlines

Fixes some linting errors and styling issues,
as well as type annotations and missing class names
from import statements.
